SonarQube.TeamBuild.PreProcessor/Interfaces/ISonarQubeServerFactory.cs

```diff
@@ -0,0 +1,18 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="ISonarQubeServerFactory.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+namespace SonarQube.TeamBuild.PreProcessor
+{
+    public interface ISonarQubeServerFactory
+    {
+        /// <summary>
+        /// Creates and returns the component that interacts with the SonarQube server
+        /// </summary>
+        /// <param name="args">Validated arguments</param>
+        ISonarQubeServer Create(ProcessedArgs args);
+    }
```

Comment:
We can't just create and pass a server instance to the TeamBuildPreProcessor as we need the pre-processor to validate the arguments first. So, we pass the pre-processor a factory it can call once it has validate the arguments.